### ISSUE TITLE
Can't get --no-pool-redirect to work with OpenWrt (compiled from latest src)

### ISSUE THREAD

**Issue by bobdole4prez** [luke-jr/bfgminer#439]
Hi luke-jr,
I compiled the latest commit in the hopes of having access to the --no-pool-redirect option, but it seems to get removed from the code during the build process.  The OPT_WITHOUT_ARG block for this option in the miner.c file is missing from the copy that gets put in the build_dir.  I'm rather at a loss as to what's happening.

Also, compilation fails if I disable the USB support suboption in make menuconfig.  It complains that libusb is missing, even when it is selected and compiled.

Thanks!

**Comment by luke-jr** [comment]
I have no idea what you're doing, but there is no build_dir, nor make menuconfig...

**Comment by bobdole4prez** [comment]
Thanks for the prompt response.  I'm trying to cross-compile bfgminer for my OpenWrt router within the OpenWrt Buildroot environment.  I'm following the instructions on the OpenWrt wiki on how to build custom packages.  The "make menuconfig" comes from the Buildroot environment.  It lets you configure what target hardware you are building for as well as what packages to include.  (According to the OpenWrt wiki, you have to select your custom package as well as its dependencies, or when you issue the make commands for that package Buildroot will just do nothing).

After running "make menuconfig", I run "make package/bfgminer/compile" from the main Buildroot directory.  This seems to be the command that generates the "build_dir", in which the code is actually compiled, and finally an .ipk package is produced.  (And the built package does work on my router.  It just doesn't recognize the --no-pool-redirect option, which was my entire reason for compiling from source).

Is there a different way to compile the OpenWrt packages?  How are you doing it?

Thanks!

**Comment by luke-jr** [comment]
Oh, to do OpenWrt packages you need to build a source file and edit the filename the OpenWrt package uses...

**Comment by bobdole4prez** [comment]
Are there any instructions available on how to do this?  Is it a
question of a few standard commands and edits?  Or is it a rather
complicated process?  I'm not overly familiar with the details of the
"make" process (I'm not an experienced programmer.  I've just built the
occasional package from source for Debian).

It'd be really nice to get the --no-pool-redirect option because my
miners were hijacked yesterday using this exploit.

Thanks for your help.

**Comment by luke-jr** [comment]
See the make-release and openwrt/multibuild.sh scripts in the BFGMiner source.

**Comment by bobdole4prez** [comment]
I ended up just modifying the BFGMiner OpenWRT Makefile to point to your
Git repository and then it worked.  I had downloaded the Git repo by
hand to openwrt_buildroot/package/BFGMiner thinking it would
automatically grab the code from there, but all it used from that folder
was the Makefile which directed it to get the code from a tarball on
your website of an older revision of the 3.10 code...  Hence why the
--no-pool-redirect option appeared to be disappearing from the source code.

Definitely my mistake!  Thanks for your help, and for all the good work!
